planet/clients/mosaics.py

```diff
@@ -0,0 +1,397 @@
+            resp = await self._quads_bbox(mosaic, search, minimal, summary)
+        # kinda yucky - yields a different "shaped" dict
+        if summary:
+            yield resp.json()["summary"]
+            return
+        async for item in MosaicQuads(resp, self._session.request):
+            yield item
+
+    async def _quads_geometry(self,
+                              mosaic: dict,
+                              geometry: dict,
+                              minimal: bool,
+                              summary: bool) -> Response:
+        params = {}
+        if minimal:
+            params["minimal"] = "true"
+        if summary:
+            params["summary"] = "true"
+        mosaic_id = mosaic["id"]
+        return await self._session.request(
+            method="POST",
+            url=self._url(f"mosaics/{mosaic_id}/quads/search"),
+            params=params,
+            json=geometry,
+        )
+
+    async def _quads_bbox(self,
+                          mosaic: dict,
+                          bbox: BBox,
+                          minimal: bool,
+                          summary: bool) -> Response:
+        quads_template = mosaic["_links"]["quads"]
+        # this is fully qualified URL, so don't use self._url
+        url = quads_template.replace("{lx},{ly},{ux},{uy}",
+                                     ",".join([str(f) for f in bbox]))
+        # params will overwrite the templated query
+        if minimal:
+            url += "&minimal=true"
+        if summary:
+            url += "&summary=true"
+        return await self._session.request(
+            method="GET",
+            url=url,
+        )
+
+    async def get_quad(self, name_or_id: str, quad_id: str) -> dict:
+        """
+        Get a mosaic's quad information.
+
+        Example:
+
+        ```
+        quad = await client.get_quad("d5098531-aa4f-4ff9-a9d5-74ad4a6301e5", "1234-5678")
+        print(quad)
+        ```
+        """
+        if not _is_uuid(name_or_id):
+            mosaic = await self.get_mosaic(name_or_id)
+            if mosaic is None:
+                return {}
+            name_or_id = cast(str, mosaic["id"])
+        resp = await self._session.request(
+            method="GET",
+            url=self._url(f"mosaics/{name_or_id}/quads/{quad_id}"),
+        )
+        return resp.json()
+
+    async def get_quad_contributions(self, quad: dict) -> list[dict]:
+        """
+        Get a mosaic's quad information.
+
+        Example:
+
+        ```
+        quad = await client.get_quad("d5098531-aa4f-4ff9-a9d5-74ad4a6301e5", "1234-5678")
+        contributions = await client.get_quad_contributions(quad)
+        print(contributions)
+        ```
+        """
+        resp = await self._session.request(
+            "GET",
+            quad["_links"]["items"],
+        )
+        return resp.json()["items"]
+
+    async def download_quad(self,
+                            quad: dict,
+                            directory,
```

Comment:
In the Data API client, the `directory` arg defaults to current dir. Is it worth making that consistent? I also think your idea of defaulting to the mosaic name would work